src/foundry/client-esm/dice/terms/dice.d.mts

```diff
@@ -0,0 +1,398 @@
+   *                    (default: `{}`)
+   */
+  constructor(termData?: Partial<DiceTerm.TermData>);
+
+  get method(): string; // TODO: Convert to CONFIG.Dice.fulfillment.methods
+
+  set method(method: string); // TODO: Convert to CONFIG.Dice.fulfillment.methods
+
+  #method: string; // TODO: Convert to CONFIG.Dice.fulfillment.methods
+
+  /**  An Array of dice term modifiers which are applied. */
+  modifiers: DiceTerm.TermData["modifiers"];
+
+  /** The array of dice term results which have been rolled. */
+  results: DiceTerm.Result[];
+
+  /* -------------------------------------------- */
+
+  /**
+   * Define the denomination string used to register this DiceTerm type in CONFIG.Dice.terms
+   * @defaultValue ""
+   */
+  static DENOMINATION: string;
+
+  /**
+   * Define the named modifiers that can be applied for this particular DiceTerm type.
+   * @defaultValue `{}`
+   */
+  static MODIFIERS: DiceTerm.Modifiers;
+
+  /**
+   * A regular expression pattern which captures the full set of term modifiers
+   * Anything until a space, group symbol, or arithmetic operator
+   */
+  static MODIFIERS_REGEXP_STRING: string;
+
+  /** A regular expression used to separate individual modifiers */
+  static MODIFIER_REGEXP: RegExp;
+
+  static REGEXP: RegExp;
+
+  /**
+   * @defaultValue `["number", "faces", "modifiers", "results", "method"]`
+   */
+  static SERIALIZE_ATTRIBUTES: string[];
+
+  /* -------------------------------------------- */
+  /*  Dice Term Attributes                        */
+  /* -------------------------------------------- */
+
+  /** The number of dice of this term to roll. Returns undefined if the number is a
+   * complex term that has not yet been evaluated.
+   */
+  get number(): DiceTerm.TermData["number"] | undefined;
+
+  /**
+   * The number of dice of this term to roll, before modifiers are applied, or a Roll instance that will be evaluated to
+   * a number.
+   */
+  _number: DiceTerm.TermData["number"] | Roll;
+
+  /** The number of dice of this term to roll. */
+  set number(value: DiceTerm.TermData["number"]);
+
+  /**
+   * The number of faces on the die. Returns undefined if the faces are represented as a complex term that has not yet
+   * been evaluated.
+   */
+  get faces(): DiceTerm.TermData["faces"] | undefined;
+
+  /** The number of faces on the die, or a Roll instance that will be evaluated to a number. */
+  _faces: DiceTerm.TermData["faces"] | Roll;
+
+  set faces(value: DiceTerm.TermData["faces"]);
+
+  /* -------------------------------------------- */
+  /**  The denomination of this DiceTerm instance. */
+  get denomination(): string;
+
+  /** An array of additional DiceTerm instances involved in resolving this DiceTerm. */
+  get dice(): DiceTerm[];
+
+  override get total(): number | undefined;
+
+  /** Return an array of rolled values which are still active within this term */
+  get values(): number[];
+
+  override get isDeterministic(): false;
```

Comment:
So this probably won't happen very often but if someone wants to extend `DiceTerm` then they can't make it non-deterministic.

Also in general any user that relies on `DiceTerm#isDeterministic` to be exactly `false` is probably doing something strange. It's therefore my _opinion_ that it should be set to `boolean`.